### ISSUE TITLE
Unable to initialize rmf_fleet_adapter

### ISSUE THREAD

**Issue by destkk** [osrf/rmf_core#266]
Hi everyone. Does anyone what are the possible causes that results in not initializing rmf_fleet_adapter? I am able to execute the launch file and opened up Gazebo and RViz but it shows the following results below,

`[ERROR] [full_control-17]: process has died [pid 21663, exit code -11, cmd '/hom/rmf_demos_ws/install/rmf_fleet_adapter/lib/rmf_fleet_adapter/full_control --ros-args -r __node:=deliveryRobot_fleet_adapter --params-file /tmp/launch_params_0egknyqn --params-file /tmp/launch_params_wmc47pt3 --params-file /tmp/launch_params_i9ovtjzf --params-file /tmp/launch_params_e4mqcpom --params-file /tmp/launch_params_ega3xqll --params-file /tmp/launch_params_q45ajgjg --params-file /tmp/launch_params_7bi_6n2s --params-file /tmp/launch_params_kmsrl3f3 --params-file /tmp/launch_params_zrhi1zlf --params-file /tmp/launch_params_9mud_zqv --params-file /tmp/launch_params_0uvxclok --params-file /tmp/launch_params_n_1_ygsq --params-file /tmp/launch_params_oyi3qjz0 --params-file /tmp/launch_params_oj85aiqr --params-file /tmp/launch_params_2wgtqgj_ --params-file /tmp/launch_params_sndybptp --params-file /tmp/launch_params_z6dwyxny --params-file /tmp/launch_params_p_db5xje --params-file /tmp/launch_params_b2skfg4t --params-file /tmp/launch_params_gdejt6wl --params-file /tmp/launch_params_6j65k5vz --params-file /tmp/launch_params_u3vunadu --params-file /tmp/launch_params_zcyw14xf --params-file /tmp/launch_params_i77ptpgq --params-file /tmp/launch_params_zz8ro5i4 --params-file /tmp/launch_params_rs3huvud --params-file /tmp/launch_params_2fni1w3y --params-file /tmp/launch_params_9nf7r3s1'].`

`[ERROR] [full_control-15]: process has died [pid 21656, exit code -11, cmd '/home/rmf_demos_ws/install/rmf_fleet_adapter/lib/rmf_fleet_adapter/full_control --ros-args -r __node:=tinyRobot_fleet_adapter --params-file /tmp/launch_params_ryg_3m77 --params-file /tmp/launch_params_7mcf94k_ --params-file /tmp/launch_params_i_v4xw9c --params-file /tmp/launch_params_tp59exre --params-file /tmp/launch_params_ly4alid6 --params-file /tmp/launch_params_fk651h4w --params-file /tmp/launch_params_n2d1o9na --params-file /tmp/launch_params_pna8ioh4 --params-file /tmp/launch_params_imie_g5z --params-file /tmp/launch_params_8407eodh --params-file /tmp/launch_params_p1a58gje --params-file /tmp/launch_params_wa2rk5qi --params-file /tmp/launch_params_fnm1goyz --params-file /tmp/launch_params_nstcdjuz --params-file /tmp/launch_params_oefq2_kw --params-file /tmp/launch_params_z_xifcbb --params-file /tmp/launch_params___9hca8u --params-file /tmp/launch_params__c6g30lt --params-file /tmp/launch_params_kd8rb9en --params-file /tmp/launch_params_ad1qrcyh --params-file /tmp/launch_params__0xzwrxr --params-file /tmp/launch_params_3w33cchz --params-file /tmp/launch_params_2_vxas1m --params-file /tmp/launch_params_of9vzngw --params-file /tmp/launch_params_ckug_orm --params-file /tmp/launch_params_4ak9si7y --params-file /tmp/launch_params_piwvg7b7 --params-file /tmp/launch_params_dw9tfshp'].`

**Comment by mxgrey** [comment]
Without knowing what you're trying to launch or how you're trying to launch it, we can't really guess at what might be wrong.

Please provide the following details:
* What system you're running on (e.g. Ubuntu 20.04)
* What branch of each repo you're using
* What command you are passing into the terminal to perform the launch

**Comment by destkk** [comment]
> Without knowing what you're trying to launch or how you're trying to launch it, we can't really guess at what might be wrong.
> 
> Please provide the following details:
> 
>     * What system you're running on (e.g. Ubuntu 20.04)
> 
>     * What branch of each repo you're using
> 
>     * What command you are passing into the terminal to perform the launch

System running on Ubuntu 20.04
Command used is `ros2 launch demos test_building.launch.xml `
I have drawn a map from traffic editor and generated both the map and navigation.

**Comment by mxgrey** [comment]
I assume `test_building.launch.xml` is a launch file that you wrote yourself. In that case, would you be able to share it with us? There still isn't much we can conclude without knowing what the launch file contains.

**Comment by destkk** [comment]
> I assume `test_building.launch.xml` is a launch file that you wrote yourself. In that case, would you be able to share it with us? There still isn't much we can conclude without knowing what the launch file contains.

Hi there. My apologies as I am still new in this. Please find the codes below,

```
<?xml version='1.0' ?>

<launch>
  <arg name="use_ignition" default="0"/>
  <arg name="gazebo_version" default="11"/>

  <!-- Common launch -->
  <include file="$(find-pkg-share demos)/common.launch.xml">
    <arg name="use_sim_time" value="true"/>
    <arg name="viz_config_file" value ="$(find-pkg-share demos)/include/test_building/test_building.rviz"/>
    <arg name="config_file" value="$(find-pkg-share rmf_demo_maps)/test_building/test_building_1.building.yaml"/>
  </include>

  <!-- Simulation launch -->
  <include file="$(find-pkg-share demos)/simulation.launch.xml">
    <arg name="map_name" value="test_building" />
    <arg name="use_ignition" value="$(var use_ignition)" />
    <arg name="gazebo_version" value="$(var gazebo_version)" />
  </include>

<!-- TinyRobot fleet adapter and robot state aggregator needed for the TinyRobot slotcar_plugin -->
  <group>
    <let name="fleet_name" value="tinyRobot"/>
    <include file="$(find-pkg-share demos)/include/adapters/tinyRobot_adapter.launch.xml">
      <arg name="fleet_name" value="$(var fleet_name)"/>
      <arg name="use_sim_time" value="$(var use_sim_time)"/>
      <arg name="nav_graph_file" value="$(find-pkg-share rmf_demo_maps)/maps/test_building/nav_graphs/0.yaml" />
    </include>
    <include file="$(find-pkg-share rmf_fleet_adapter)/robot_state_aggregator.launch.xml">
      <arg name="robot_prefix" value="tinyRobot"/>
      <arg name="fleet_name" value="$(var fleet_name)"/>
      <arg name="use_sim_time" value="true"/>
    </include>
  </group>
  
<!-- DeliveryRobot fleet adapter and robot state aggregator needed for DeliveryRobot slotcar_plugin -->
  <group>
    <let name="fleet_name" value="deliveryRobot"/>
    <include file="$(find-pkg-share demos)/include/adapters/deliveryRobot_adapter.launch.xml">
      <arg name="fleet_name" value="$(var fleet_name)"/>
      <arg name="use_sim_time" value="$(var use_sim_time)"/>
      <arg name="nav_graph_file" value="$(find-pkg-share rmf_demo_maps)/maps/test_building/nav_graphs/1.yaml" />
    </include>
    <include file="$(find-pkg-share rmf_fleet_adapter)/robot_state_aggregator.launch.xml">
      <arg name="robot_prefix" value="deliveryRobot"/>
      <arg name="fleet_name" value="$(var fleet_name)"/>
      <arg name="use_sim_time" value="true"/>
    </include>
  </group>


</launch>
```
The codes are copied from the other reference file (hotel.launch.xml)
I have named both robots according to the screenshots shown below,

![Screenshot from 2021-01-20 14-28-00](https://user-images.githubusercontent.com/63645940/105135971-d29a9700-5b2b-11eb-8e9f-ff238e2a36c5.png)
![Screenshot from 2021-01-20 14-28-02](https://user-images.githubusercontent.com/63645940/105135977-d3cbc400-5b2b-11eb-82ef-9e62bcf9db96.png)

**Comment by Achllle** [comment]
I'm running into the same issue, code `11` means an invalid memory reference I'm assuming somewhere in [full_control](https://github.com/osrf/rmf_core/blob/master/rmf_fleet_adapter/src/full_control/main.cpp). I'm able to run the default office scenario, but as soon as I plug in custom files from traffic editor, this node crashes. Unlike the OP I'm using real robots (which aren't showing up in RViz although the map and traffic lanes are)

@mxgrey adding in some relevant info to add on to this issue:

* sample message on `/fleet_states` (I'm not sure how to change the level_name in free_fleet so I changed everything to be the default...)
```
name: fetch
robots:
- name: fetch26
  model: robot_model
  task_id: ''
  seq: 0
  mode:
    mode: 0
    mode_request_id: 0
  battery_percent: 100.0
  location:
    t:
      sec: 1616633676
      nanosec: 857054821
    x: -0.09609447419643402
    y: -8.53885269165039
    yaw: -1.4772193431854248
    level_name: level_name
    index: 0
  path: []
```
* My core + adapter [launch file](https://github.com/Achllle/rmf_demos/blob/ff_demo/demos/launch/ff.launch.xml)
* My fleet adapter [launch file](https://github.com/Achllle/rmf_demos/blob/ff_demo/demos/launch/include/adapters/fetch_adapter.launch.xml)
* [map directory](https://github.com/Achllle/rmf_demos/tree/ff_demo/rmf_demo_maps/maps)

I'm assuming what's happening is that some field is empty or incorrect somewhere causing the full_control node to crash. Either way the error is quite cryptic and probably should be caught. Any pointers appreciated!

**Comment by mxgrey** [comment]
There is a PR [here](https://github.com/open-rmf/rmf_ros2/pull/13) which addresses a known memory issue that could be triggered close to startup when trying to run a scenario that doesn't meet the fleet adapter's expectations. I suggest giving that a try.

I'll see if I can reproduce the issue on my end using the files that you've provided. Hopefully I'll have time to do that before the end of tomorrow.

As a meta remark, we're migrating over to the https://github/open-rmf organization, and the meta repo's [discussion board](https://github.com/open-rmf/rmf/discussions) is replacing this one.

**Comment by Achllle** [comment]
Can confirm that's the issue. I know get the error message described in that PR now. Not sure which scenario is causing it though.
I think this can be closed.

**Comment by mxgrey** [comment]
Closed by open-rmf/rmf_ros2#13
